Linux GTK integration build error

<!--
Thanks for reporting issues of Telegram Desktop!

To make it easier for us to help you please enter detailed information below.
-->
### Steps to reproduce
1. Build tdesktop
2. 
3. 

**Logs**:
```
/var/tmp/portage/net-im/telegram-desktop-2.5.6/work/tdesktop-2.5.6-full/Telegram/SourceFiles/platform/linux/linux_gtk_integration.cpp:331:3: error: assigning to 'GdkScreen *(*)(GdkDisplay *)' (aka '_GdkScreen *(*)(_GdkDisplay *)') from incompatible type 'GdkMonitor *(GdkDisplay *)' (aka '_GdkMonitor *(_GdkDisplay *)'): different return type ('GdkScreen *' (aka '_GdkScreen *') vs 'GdkMonitor *' (aka '_GdkMonitor *'))
                LOAD_GTK_SYMBOL(lib_gtk, "gdk_display_get_primary_monitor", gdk_display_get_primary_monitor);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/net-im/telegram-desktop-2.5.6/work/tdesktop-2.5.6-full/Telegram/SourceFiles/platform/linux/linux_gtk_integration_p.h:24:50: note: expanded from macro 'LOAD_GTK_SYMBOL'
#define LOAD_GTK_SYMBOL(lib, name, func) (func = ::func)
                                                 ^~~~~~
/var/tmp/portage/net-im/telegram-desktop-2.5.6/work/tdesktop-2.5.6-full/Telegram/SourceFiles/platform/linux/linux_gtk_integration.cpp:332:3: error: assigning to 'int (*)(GdkScreen *)' (aka 'int (*)(_GdkScreen *)') from incompatible type 'int (GdkMonitor *)' (aka 'int (_GdkMonitor *)'): type mismatch at 1st parameter ('GdkScreen *' (aka '_GdkScreen *') vs 'GdkMonitor *' (aka '_GdkMonitor *'))
                LOAD_GTK_SYMBOL(lib_gtk, "gdk_monitor_get_scale_factor", gdk_monitor_get_scale_factor);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/net-im/telegram-desktop-2.5.6/work/tdesktop-2.5.6-full/Telegram/SourceFiles/platform/linux/linux_gtk_integration_p.h:24:50: note: expanded from macro 'LOAD_GTK_SYMBOL'
#define LOAD_GTK_SYMBOL(lib, name, func) (func = ::func)
                                                 ^~~~~~
2 errors generated.
```

--- a/Telegram/SourceFiles/platform/linux/linux_gtk_integration_p.h
+++ b/Telegram/SourceFiles/platform/linux/linux_gtk_integration_p.h
@@ -27,7 +27,7 @@ extern "C" {
 #endif // !LINK_TO_GTK
 
 // To be able to compile with gtk-2.0 headers as well
-#define GdkMonitor GdkScreen
+typedef struct _GdkMonitor GdkMonitor;
 typedef struct _GtkAppChooser GtkAppChooser;
 
 namespace Platform {

--- a/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
@@ -718,39 +718,42 @@ void Create(Window::Notifications::System *system) {
 		}
 	};
 
+	if (!ServiceRegistered) {
+		CurrentServerInformation = std::nullopt;
+		CurrentCapabilities = QStringList{};
+		InhibitionSupported = false;
+		managerSetter();
+		return;
+	}
+
+	// There are some asserts that manager is not nullptr,
+	// avoid crashes until some real manager is created
 	if (!system->managerType().has_value()) {
 		using DummyManager = Window::Notifications::DummyManager;
 		system->setManager(std::make_unique<DummyManager>(system));
 	}
 
-	if (ServiceRegistered) {
-		const auto counter = std::make_shared<int>(3);
-		const auto oneReady = [=] {
-			if (!--*counter) {
-				managerSetter();
-			}
-		};
+	const auto counter = std::make_shared<int>(3);
+	const auto oneReady = [=] {
+		if (!--*counter) {
+			managerSetter();
+		}
+	};
 
-		GetServerInformation([=](std::optional<ServerInformation> result) {
-			CurrentServerInformation = result;
-			oneReady();
-		});
+	GetServerInformation([=](std::optional<ServerInformation> result) {
+		CurrentServerInformation = result;
+		oneReady();
+	});
 
-		GetCapabilities([=](QStringList result) {
-			CurrentCapabilities = result;
-			oneReady();
-		});
+	GetCapabilities([=](QStringList result) {
+		CurrentCapabilities = result;
+		oneReady();
+	});
 
-		GetInhibitionSupported([=](bool result) {
-			InhibitionSupported = result;
-			oneReady();
-		});
-	} else {
-		CurrentServerInformation = std::nullopt;
-		CurrentCapabilities = QStringList{};
-		InhibitionSupported = false;
-		managerSetter();
-	}
+	GetInhibitionSupported([=](bool result) {
+		InhibitionSupported = result;
+		oneReady();
+	});
 }
 
 class Manager::Private {

--- a/Telegram/build/docker/centos_env/Dockerfile
+++ b/Telegram/build/docker/centos_env/Dockerfile
@@ -255,6 +255,7 @@ RUN git clone -b 2.10.0 --depth=1 $GIT/intel/libva.git
 WORKDIR libva
 RUN ./autogen.sh \
 	--enable-static \
+	--sysconfdir=/etc \
 	--with-drivers-path=/usr/lib/dri
 
 RUN make -j$(nproc)

